server/fleet/microsoft_mdm.go

```diff
@@ -380,10 +416,64 @@ type ContextItem struct {
 }
 
 type AdditionalContext struct {
-	XMLNS       string        `xml:"xmlns,attr"`
-	ContextItem []ContextItem `xml:"ContextItem"`
+	XMLNS        string        `xml:"xmlns,attr"`
+	ContextItems []ContextItem `xml:"ContextItem"`
+}
+
+// Get Binary Security Token
+func (msg RequestSecurityToken) GetBinarySecurityTokenData() (string, error) {
+	if len(msg.BinarySecurityToken.Content) == 0 {
+		return "", errors.New("BinarySecurityToken is empty")
+	}
+
+	return msg.BinarySecurityToken.Content, nil
+}
+
+// Get Binary Security Token Type
+func (msg RequestSecurityToken) GetBinarySecurityTokenType() (BinSecTokenType, error) {
+	if msg.BinarySecurityToken.ValueType == mdm.EnrollReqTypePKCS10 {
+		return MDETokenPKCS10, nil
+	}
+
+	if msg.BinarySecurityToken.ValueType == mdm.EnrollReqTypePKCS7 {
+		return MDETokenPKCS7, nil
+	}
+
+	return MDETokenPKCSInvalid, errors.New("BinarySecurityToken is invalid")
+}
+
+// Get Context Item
+func (msg RequestSecurityToken) GetContextItem(item string) (string, error) {
+	if len(msg.AdditionalContext.ContextItems) == 0 {
+		return "", errors.New("ContextItems is empty")
+	}
+
+	// Generate map of ContextItems if not there
+	if msg.MapContextItems == nil {
+		contextMap := make(map[string]ContextItem)
+		for _, item := range msg.AdditionalContext.ContextItems {
+			contextMap[item.Name] = item
+		}
+		msg.MapContextItems = &contextMap
```

Comment:
Currently this is probably executed every time because the method is defined on a non-pointer receiver (`(msg RequestSecurityToken)`), so the fields assigned inside the method have no effect on the caller. This should be changed to `(msg *RequestSecurityToken)` to be used as cache (to see the already filled map on subsequent calls).